scalyr_agent/builtin_monitors/thread_pool.py

```diff
@@ -0,0 +1,59 @@
+import threading
+from concurrent.futures import ThreadPoolExecutor
+
+from scalyr_agent import scalyr_logging
+
+global_log = scalyr_logging.getLogger(__name__)
+
+class ExecutorMixIn:
+    def __init__(self):
+        # Using 4 threads for processing requests, because of GIL and CPU bound tasks higher number would not help process the data faster.
+        self._request_processing_executor = ThreadPoolExecutorFactory.get_singleton("request_processing_executor", max_workers=4)

```

Comment:
Do we want to make `max_workers` configurable?

Perhaps it's a good idea in case someone hits a race condition or similar so we can provide them with a workaround to set this value to `1`.

Reply:
This was a laziness on my part, I will make it configurable :)